apps/examples/testcase/ta_tc/eventloop/utc/utc_eventloop_main.c

```diff
@@ -0,0 +1,493 @@
+	timer = eventloop_add_timer(5000, false, (timeout_callback)repeat_timer_stop_cb, EL_REPEAT_TIMER_DATA);
+	TC_ASSERT_NEQ("eventloop_add_timer", timer, NULL);
+	ret = eventloop_loop_run();
+	TC_ASSERT_EQ("eventloop_loop_run", ret, OK);
+	TC_ASSERT_EQ("eventloop_add_timer", el_timer_flag, 4);
+
+	/* Add timer which repeats forever, it is stooped by calling eventloop_loop_stop. */
+	el_timer_flag = 0;
+	repeat_timer = eventloop_add_timer(2000, true, repeat_timer_cb, EL_REPEAT_TIMER_DATA);
+	TC_ASSERT_NEQ("eventloop_add_timer", repeat_timer, NULL);
+	loop_stop_timer = eventloop_add_timer(3000, false, (timeout_callback)loop_stop_cb, NULL);
+	TC_ASSERT_NEQ("eventloop_add_timer", loop_stop_timer, NULL);
+	ret = eventloop_loop_run();
+	TC_ASSERT_EQ("eventloop_loop_run", ret, OK);
+	TC_ASSERT_EQ("eventloop_add_timer", el_timer_flag, 1);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_delete_timer_n(void)
+{
+	int ret;
+
+	ret = eventloop_delete_timer(NULL);
+	TC_ASSERT_EQ("eventloop_delete_timer", ret, EVENTLOOP_INVALID_PARAM);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_delete_timer_p(void)
+{
+	el_timer_t *timer;
+	int ret;
+
+	timer = eventloop_add_timer(0, false, (timeout_callback)timer_cb, NULL);
+	TC_ASSERT_NEQ("eventloop_add_timer", timer, NULL);
+	ret = eventloop_delete_timer(timer);
+	TC_ASSERT_EQ("eventloop_delete_timer", ret, OK);
+
+	/* Try to delete already deleted timer, it will be return EVENTLOOP_INVALID_HANDLE. */
+	ret = eventloop_delete_timer(timer);
+	TC_ASSERT_EQ("eventloop_delete_timer", ret, EVENTLOOP_INVALID_HANDLE);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_async_n(void)
+{
+	el_timer_t *timer;
+
+	timer = eventloop_add_timer_async(1000, false, NULL, NULL);
+	TC_ASSERT_EQ("eventloop_add_timer_async", timer, NULL);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_eventloop_add_timer_async_p(void)
+{
+	int sleep_cnt;
+	el_timer_t *timer;
+	el_timer_t *repeat_timer;
+
+	el_timer_flag = 0;
+	timer = eventloop_add_timer_async(0, false, (timeout_callback)timer_cb, EL_TIMER_DATA);
+	TC_ASSERT_NEQ("eventloop_add_timer_async", timer, NULL);
+	sleep_cnt = 0;
+	while (sleep_cnt < 5) {
+		sleep(1);
+		sleep_cnt++;
+		if (el_timer_flag == 1) {
+			break;
+		}
+	}
+	TC_ASSERT_EQ("eventloop_add_timer_async", el_timer_flag, 1);
+
+	el_timer_flag = 0;
+	repeat_timer = eventloop_add_timer_async(1000, true, repeat_timer_3times_cb, EL_REPEAT_TIMER_DATA);
+	TC_ASSERT_NEQ("eventloop_add_timer_async", repeat_timer, NULL);
+	sleep_cnt = 0;
+	while (sleep_cnt < 5) {
+		sleep(1);
+		sleep_cnt++;
+		if (el_timer_flag == 3) {
+			break;
+		}
+	}
+	TC_ASSERT_EQ("eventloop_add_timer_async", el_timer_flag, 3);
+
+	TC_SUCCESS_RESULT();
+}
+
+static int wifi_on_callback(void *cb_data, void *event_data)
+{
+	static int cnt = 0;
+
+	printf("WIFI ON CALLBACK!! %d\n", ++cnt);
+
+	if (cb_data == NULL && event_data != NULL && strncmp(event_data, EL_WIFI_ON_DATA, sizeof(EL_WIFI_ON_DATA) + 1) == 0) {
```

Comment:
for strncmp, sizeof(EL_WIFI_ON_DATA) + 1 is correct? not sizeof(EL_WIFI_ON_DATA) or strlen(EL_WIFI_ON_DATA) + 1?